src/main/java/com/aws/greengrass/deployment/DeviceConfiguration.java

```diff
@@ -776,7 +776,7 @@ private void validateDeviceConfiguration(String thingName, String certificateFil
         if (Utils.isEmpty(thingName)) {
             errors.add(DEVICE_PARAM_THING_NAME + CANNOT_BE_EMPTY);
         }
-        if (Utils.isEmpty(certificateFilePath)) {
+        if (!Utils.isPkcs11Uri(privateKeyPath) && Utils.isEmpty(certificateFilePath)) {
```

Comment:
Nucleus *must not* be aware of PKCS11

Reply:
We can rely on security service to do the right thing, just make it optional here?